src/components/item/form/DocumentForm.tsx

```diff
@@ -44,29 +62,40 @@ export const DocumentExtraForm = ({
   showActions?: boolean;
 }): JSX.Element => {
   const { t } = useBuilderTranslation();
+  const [editorMode, setEditorMode] = useState(EditorMode.Rich);
   const flavorsTranslations = Object.values(DocumentItemExtraFlavor).map(
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     (f) => [f, t(BUILDER[`DOCUMENT_FLAVOR_${f.toUpperCase()}`])],
   );
 
+  const withFlavor = (textView: JSX.Element): JSX.Element =>
+    extra?.flavor ? (
+      <Alert severity={extra.flavor}>{textView}</Alert>
+    ) : (
+      textView
+    );
+
+  const handleChangeEditorMode = (mode: string) => {
```

Comment:
```suggestion
  const handleChangeEditorMode = (mode: EditorMode | `${EditorMode}`) => {
    setEditorMode(EditorMode[EditorMode[mode] as keyof typeof EditorMode]);
  }
```

Maybe this could solve as unknown but still have to cast.

Reply:
the issue is that the onChange method from the tabs does not return the enum type, so we can't make the typing work here. I opted to use a simple string instead. It is a bit ugly but in the end in JS it will not be safer.